include/sound/sof/ipc4/header.h

```diff
@@ -0,0 +1,333 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause) */
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2021 Intel Corporation. All rights reserved.
+ */
+
+#ifndef __INCLUDE_SOUND_SOF_IPC4_HEADER_H__
+#define __INCLUDE_SOUND_SOF_IPC4_HEADER_H__
+
+#include <linux/types.h>
+#include <uapi/sound/sof/abi.h>
+
+/** \addtogroup sof_uapi uAPI
+ *  SOF uAPI specification.
+ *  @{
+ */
+
+/*
+ * IPC4 messages have two 32 bit identifier made up as follows :-
+ *
+ * header - msg type, msg id, msg direction ...
+ * extension - extra params such as msg data size in mailbox
+ *
+ * These are sent at the start of the IPC message in the mailbox. Messages
+ * should not be sent in the doorbell (special exceptions for firmware).
+ */
+
+/*
+ * IPC4 has a 32 bits header, and the bit allocation is
+ * rsvd0 : 24;
+ * type : 5;
+ * rsp : 1;
+ * msg_tgt : 1;
+ * _reserved_0 : 1;
+ *
+ * where bits 24 .. 31 are used globally, others (rsvd0) will
+ * have different definitions for each message type.
+ */
```

Comment:
I don't know how to decode this...
Are you tryi to tell that in the 32bit header:
```
bit 0-23:	message type specific
bit 24-28:	type
bit 29:		rsp (??)
bit 30:		msg_tgt (target?)
bit 31:		reserved, unused
```

Reply:
> I don't know how to decode this...
> Are you tryi to tell that in the 32bit header:
> 
> ```
> bit 0-23:	message type specific
> bit 24-28:	type
> bit 29:		rsp (??)
> bit 30:		msg_tgt (target?)
> bit 31:		reserved, unused
> ```

Yes, exactly. The `rsp` value is either SOF_IPC4_MSG_REQUEST or SOF_IPC4_MSG_REPLY